akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -301,6 +310,8 @@ import akka.stream.scaladsl.Source
 
     private val streamDone = source.run()
 
+    streamDone.onComplete(_ => statusObserver.stopped(projectionId))(system.executionContext)
```

Comment:
Here we will have the same problem we had with the stop hook when testing with a SinkProbe. 

We should add those hooks around the `mappedSource` instead.

Reply:
Good catch. I wanted to avoid calling them when restarted, but doesn't matter much so I have changed so they are called when restarted also. Same as we have for the handler lifecycle. Fixed in 12cfdfb